src/AppSystem/Background/BackgroundApp.vala

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ *
+ * Authored by: Leonhard Kargl <leo.kargl@proton.me>
+ */
+
+public class Dock.BackgroundApp : Object {
+    public DesktopAppInfo app_info { get; construct; }
+    public Icon icon { get { return app_info.get_icon (); } }
+    public string? instance { get; construct; }
+    public string? message { get; construct; }
+
+    public BackgroundApp (DesktopAppInfo app_info, string? instance, string? message) {
+        Object (app_info: app_info, instance: instance, message: message);
+    }
+
+    public async void kill () throws Error {
+        var app_id = remove_desktop_suffix (app_info.get_id ());
+
+        try {
+            var object_path = "/" + app_id.replace (".", "/").replace ("-", "_");
+            var parameters = new Variant (
+                "(s@av@a{sv})", "quit", new Variant.array (VariantType.VARIANT, {}),
+                new Variant.array (new VariantType.dict_entry (VariantType.STRING, VariantType.VARIANT), {})
+            );
+
+            var session_bus = yield Bus.get (SESSION, null);
+
+            yield session_bus.call (
```

Comment:
I'm guessing it was done this way because DesktopAppInfo.launch_action doesn't return errors? If that's the case maybe a comment would be nice so someone doesn't try to refactor it later